### PR TITLE
Run tests from Xcode12. Github workflow

### PR DESCRIPTION
**Pull request by 3a4oT**
## Changes in this pull request

- Added GitHub workflow `CI`.
- fixed macOS/tvOS and cocoapods/carthage build

### Checklist

- [x] All tests pass. Demo project builds and runs.
- [x] I added tests, an experiment, or detailed why my change isn't tested.
- [ ] I added an entry to the `CHANGELOG.md` for any breaking changes, enhancements, or bug fixes.
- [x] I have reviewed the [contributing guide](https://github.com/Instagram/IGListKit/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by 3a4oT** [comment]
all tests passed on my fork:
https://github.com/3a4oT/IGListKit/actions/runs/368847535

**Comment by 3a4oT** [comment]
https://github.com/Instagram/IGListKit/pull/1478

**Comment by lorixx** [comment]
Thanks @3a4oT! Can you change the description of this PR? And how can I trigger the github test? Would be great to put to command and steps in the PR description too

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+          restore-keys: |
+            ${{ runner.os }}-gems-
+
+      - name: Install ruby gems.
+        run: bundle install
+
+      - name: Run unit tests for macOS
+      #DEBUG locally
+      # xcodebuild build build-for-testing -workspace "IGListKit.xcworkspace" -scheme "IGListKit-macOS" -destination "platform=macOS" -configuration Debug ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=YES GCC_GENERATE_TEST_COVERAGE_FILES=YES ONLY_ACTIVE_ARCH=YES | bundle exec xcpretty -c
+      # xcodebuild analyze test-without-building -workspace "IGListKit.xcworkspace" -scheme "IGListKit-macOS" -destination "platform=macOS" -configuration Debug ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=YES GCC_GENERATE_TEST_COVERAGE_FILES=YES ONLY_ACTIVE_ARCH=YES | bundle exec xcpretty -c
```
Comment (lorixx) [inline .github/workflows/CI.yml:37]: Remove?

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+          restore-keys: |
+            ${{ runner.os }}-gems-
+
+      - name: Install ruby gems.
+        run: bundle install
+
+      - name: iOS - ${{ matrix.destination }}
+      #DEBUG locally
+      # xcodebuild build build-for-testing -workspace "IGListKit.xcworkspace" -scheme "IGListKit" -destination "OS=14.1,name=iPhone 12 Pro" -configuration Debug ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=YES GCC_GENERATE_TEST_COVERAGE_FILES=YES ONLY_ACTIVE_ARCH=YES | bundle exec xcpretty -c
+      # xcodebuild analyze test-without-building -workspace "IGListKit.xcworkspace" -scheme "IGListKit" -destination "OS=14.1,name=iPhone 12 Pro" -configuration Debug ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=YES GCC_GENERATE_TEST_COVERAGE_FILES=YES ONLY_ACTIVE_ARCH=YES | bundle exec xcpretty -c
```
Comment (lorixx) [inline .github/workflows/CI.yml:71]: remove ?

**Comment by 3a4oT** [comment]
Hi @lorixx. Those tests would be run automatically on each new PR or new commit on existing PR after it would be merged. Since this change is not in a repository yet I think you can try to follow[ these instructions ](https://docs.github.com/en/free-pro-team@latest/actions/managing-workflow-runs/re-running-a-workflow).  Let me know if it works! I removed some boring text from the PR description does it looks correct now?

File: .github/workflows/CI.yml (hunk 2 above)
Comment (3a4oT) [reply .github/workflows/CI.yml]: done

**Comment by 3a4oT** [comment]
https://developer.apple.com/documentation/foundation/nsindexpath says that NSIndexPath defined at Foundation framework. I'm not sure I understand the motivation of https://github.com/Instagram/IGListKit/commit/9cb16b0f31135993791a774bd4844317965fa318 but I fixed macOS and cocoapods build.

**Comment by 3a4oT** [comment]
`-[IGListAdapterUpdaterTests test_whenCollectionViewSectionCountIsIncorrect_thatDoesNotCrash]` fails. I think it's not related to this PR and can be fixed as a separate PR. cc @lorixx 

CI run: https://github.com/3a4oT/IGListKit/runs/2040554348?check_suite_focus=true

**Comment by lorixx** [comment]
Hi @3a4oT , sorry for the super late review! I am following up with this PR and hopefully we can get it merge next week. The Travis CI had been deprecated and we would love to use the Github Action! Again, thank you so much for your hard working getting us here!

**Comment by 3a4oT** [comment]
Do you need something else from me or do you just need to follow your internal process to land it?

**Comment by lorixx** [comment]
Can you trigger the test run again and make sure the CI build and test successfully? I remembered I had to resolve some compile issue when I run it locally. 

The IGListDiffKit build introduced UIKit/UIKit.h which isn't ideal, I will resolve that from the internal build.

**Pull request closed without merging**